middlewarerule.js

```diff
@@ -222,8 +222,15 @@ MiddlewareRule.prototype.getCollection = function(params, data, req) {
 	}
 
 	var nonSpecialParams = Object.keys(filteredParams).filter(function(key) {
+		// Exclude special params.
 		return specialParams.indexOf(key) === -1;
-	});
+	}).filter(function(key) {
+		// Exclude path params.
+		if (!this.path.keys) { return true; }
+		return this.path.keys.reduce(function(soFar, pathKeyInfo) {
+			return soFar && (pathKeyInfo.name === key);
+		}, true);
```

Comment:
Could this `reduce` be replaced with [`Array.prototype.every`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)?

```js
this.path.keys.every(function(pathKeyInfo) {
  return pathKeyInfo.name === key;
});
```